web/packages/teleport/src/SessionRecordings/view/player/tty/TtyPlayer.ts

```diff
@@ -0,0 +1,202 @@
+/**
+ * Teleport
+ * Copyright (C) 2025 Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { getPlatform } from '@floating-ui/react/utils';
+import { CanvasAddon } from '@xterm/addon-canvas';
+import { ImageAddon } from '@xterm/addon-image';
+import { WebLinksAddon } from '@xterm/addon-web-links';
+import { WebglAddon } from '@xterm/addon-webgl';
+import { ITerminalAddon, Terminal } from '@xterm/xterm';
+
+import { Platform } from 'design/platform';
+import { ThemeColors } from 'design/theme';
+import { getMonoFont } from 'design/theme/fonts';
+
+import { Player } from '../Player';
+import { AspectFitAddon } from './AspectFitRatio';
+import { EventType, type TtyEvent } from './types';
+
+/**
+ * TtyPlayer is a player that connects a stream of TtyEvents to an xterm.js terminal.
+ *
+ * It handles rendering the terminal, applying events, and managing terminal addons.
+ *
+ * It supports resizing, clearing the terminal, and focusing the terminal on play/seek.
+ * It also handles terminal themes and font settings.
+ */
+export class TtyPlayer extends Player<TtyEvent> {
+  private addons: ITerminalAddon[] = [];
+  private aspectFitAddon = new AspectFitAddon();
+  private terminal: Terminal | null = null;
+  private playing = false;
+
+  constructor(
+    private theme: ThemeColors['terminal'],
+    private cols: number,
+    private rows: number
+  ) {
+    super();
+  }
+
+  override init(element: HTMLElement) {
+    this.terminal = new Terminal({
+      fontSize: getPlatform() === Platform.macOS ? 12 : 14,
```

Comment:
Why is this platform-specific? And what kind of impact does it have if the terminal window is scaled anyway?

Reply:
This is how the current terminal does it - https://github.com/gravitational/teleport/blob/868fcea780967197e5ebd514190099e1c7aa988a/web/packages/teleport/src/SessionRecordings/view/Xterm/Xterm.tsx#L59-L60

Not entirely sure why, but without it the text is pretty big on Mac, can't confirm for any other platform though